walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,50 @@
 
 package com.google.sps;
 
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 
+/**
+ * Class that finds suitable times for a group of people to meet.
+ **/
 public final class FindMeetingQuery {
+  /**
+   * Returns a list of TimeRanges where a group of people can meet,
+   * restrained by the events they have throughout the day.
+   *
+   * Part the algorithm iterates through the list of events,
+   * finding potential time of meetings in-between events. The second
+   * part then goes through these potential meeting times and looks for
+   * those that satisfy the amount of time that needs to be allotted, as
+   * specified in the request.
+   *
+   * @param {Collection<Event>} events
+   * @param {MeetingRequest} request
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<Event> eventsList = new ArrayList<>(events);
+    Collection<TimeRange> timeOfMeetings = new ArrayList<>();
+    int potentialMeetingTimeStart = TimeRange.START_OF_DAY;
+    eventsList.removeIf(event -> !request.getAttendees().containsAll(event.getAttendees()));
```

Comment:
Is this correct? Suppose I want to schedule a 1:1 with Chuck, and Chuck already has a 1:1 with you. My request doesn't contain you as an attendee, so this would discard you and Chuck's 1:1, which could potentially conflict.

Reply:
That is a very good point, none of the tests covered if an event has more than one attendee, especially one that's outside the meeting request so I am going to try and test this behavior.